lib/Autocomplete.js

```diff
@@ -270,13 +284,17 @@ class Autocomplete extends React.Component {
 
     ArrowUp(event) {
       event.preventDefault()
-      const itemsLength = this.getFilteredItems().length
-      if (!itemsLength) return
+      let items = this.getFilteredItems()
+      if (!items.length) return
+
       const { highlightedIndex } = this.state
-      const index = (
-        highlightedIndex === 0 ||
-        highlightedIndex === null
-      ) ? itemsLength - 1 : highlightedIndex - 1
+      let index = highlightedIndex
+
+      for(let i = 0; i < items.length ; i++) {
+        index = (index === null || index === 0) ?  items.length - 1 : index - 1
```

Comment:
If you want to use modulo here, remember to add items.length so you don't go into the negative number range when looping around: `index = (index - 1 + items.length) % items.length`.